File: 01_Programming_Basics/_03_ComplexConditionalStatementsExercise/_04_FishingBoat.java
package _01_Programming_Basics._03_ComplexConditionalStatementsExercise;

import java.util.Scanner;

public class _04_FishingBoat {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        int budget = Integer.parseInt(scan.nextLine());
        String season = scan.nextLine();
        int ppl = Integer.parseInt(scan.nextLine());

        double price = 0;
        boolean isValid = season.equals("Autumn");

        switch (season) {
            case "Spring":
                price= 3000;
                if (ppl<=6){
                    price *=0.90;
                }else if (ppl<=11){
                    price *=0.85;
                }else{
                    price *=0.75;
                }
                break;
            case "Summer":
            case "Autumn":
                price = 4200;
                if (ppl<=6){
                    price *=0.90;
                }else if (ppl<=11){
                    price *=0.85;
                }else{
                    price *=0.75;
                }
                break;
            case "Winter":
                price = 2600;
                if (ppl<=6){
                    price *=0.90;
                }else if (ppl<=11){
                    price *=0.85;
                }else{
                    price *=0.75;
                }
                break;
        }
        if (isValid) {
            price*=1.00;
        }else {
            if (ppl % 2 == 0) {
                price *= 0.95;
            } else {
                price *= 1.00;
            }
        }
        double sum = budget - price;

        if(budget>=price){
            System.out.printf("Yes! You have %.2f leva left.", Math.abs(sum));
        }else{
            System.out.printf("Not enough money! You need %.2f leva.", Math.abs(sum));
        }
    }
}

File: 23_C#Basics/_03_ComplexConditionalStatementsExercise/_04_FishingBoat/_04_FishingBoat.cs
using System;

public class _04_FishingBoat
{
    public static void Main(string[] args)
    {
        // Read input values
        int budget = int.Parse(Console.ReadLine());
        string season = Console.ReadLine();
        int numberOfPeople = int.Parse(Console.ReadLine());

        double basePrice = 0;
        bool isAutumn = season.Equals("Autumn");

        // Calculate the base price based on the season
        // Apply discounts based on the number of people
        switch (season)
        {
            case "Spring":
                basePrice = 3000;
                if (numberOfPeople <= 6)
                {
                    basePrice *= 0.90;
                }
                else if (numberOfPeople <= 11)
                {
                    basePrice *= 0.85;
                }
                else
                {
                    basePrice *= 0.75;
                }
                break;
            case "Summer":
            case "Autumn":
                basePrice = 4200;
                if (numberOfPeople <= 6)
                {
                    basePrice *= 0.90;
                }
                else if (numberOfPeople <= 11)
                {
                    basePrice *= 0.85;
                }
                else
                {
                    basePrice *= 0.75;
                }
                break;
            case "Winter":
                basePrice = 2600;
                if (numberOfPeople <= 6)
                {
                    basePrice *= 0.90;
                }
                else if (numberOfPeople <= 11)
                {
                    basePrice *= 0.85;
                }
                else
                {
                    basePrice *= 0.75;
                }
                break;
        }

        // Apply additional adjustments based on season and odd/even number of people
        if (isAutumn)
        {
            basePrice *= 1.00;
        }
        else
        {
            if (numberOfPeople % 2 == 0)
            {
                basePrice *= 0.95;
            }
            else
            {
                basePrice *= 1.00;
            }
        }

        double difference = budget - basePrice;

        // Determine the appropriate output based on budget
        if (budget >= basePrice)
        {
            Console.WriteLine($"Yes! You have {Math.Abs(difference):F2} leva left.");
        }
        else
        {
            Console.WriteLine($"Not enough money! You need {Math.Abs(difference):F2} leva.");
        }
    }
}
